tests/test_simulation.py

```diff
@@ -454,7 +506,12 @@ def test_run(seq, patch_plt_show):
     with pytest.warns(
         DeprecationWarning, match="Setting `initial_state` is deprecated"
     ):
-        sim.initial_state = good_initial_array
+        with pytest.warns(
+            DeprecationWarning, match="The `Simulation` class is deprecated"
+        ):
+            _sim = Simulation(seq, sampling_rate=0.01)
+        _sim.initial_state = good_initial_array
+        _sim.initial_state
```

Comment:
I guess this second line is not doing anything, right?

Reply:
Then take the opportunity to do an assert
```suggestion
        _sim.initial_state
```
```suggestion
        _sim.initial_state == good_initial_array
```
You should also be able to do this outisde the `pytest.warns` block without issues